docs/pages/guides/spend-permissions/api-reference.mdx

```diff
@@ -0,0 +1,275 @@
+
+```solidity
+function approveBatchWithSignature(SpendPermissionBatch memory spendPermissionBatch, bytes calldata signature) external;
+```
+
+---
+
+#### `spend`
+
+Spend tokens using a spend permission, transferring them from the account to the spender. Only callable by the spender specified in the permission.
+
+```solidity
+function spend(SpendPermission memory spendPermission, uint160 value) external requireSender(spendPermission.spender);
+```
+
+---
+
+#### `revoke`
+
+Revoke a spend permission, permanently disabling its use. Only callable by the account owner specified in the spend permission.
+
+```solidity
+function revoke(SpendPermission calldata spendPermission) external requireSender(spendPermission.account);
+```
+
+---
+
+#### `getHash`
+
+Generate a hash of a `SpendPermission` struct for signing, in accordance with [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md).
+
+```solidity
+function getHash(SpendPermission memory spendPermission) public view returns (bytes32);
+```
+
+---
+
+#### `getBatchHash`
+
+Generate a hash of a `SpendPermissionBatch` struct for signing, in accordance with [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md). 
+Reverts if the batch is empty.
+
+```solidity
+function getBatchHash(SpendPermissionBatch memory spendPermissionBatch) public view returns (bytes32);
+```
+
+---
+
+#### `isApproved`
+
+Check if a spend permission is approved and not revoked.
+
+```solidity
+function isApproved(SpendPermission memory spendPermission) public view returns (bool);
+```
+
+---
+
+#### `getCurrentPeriod`
+
+Retrieve the start, end, and accumulated spend for the current period of a spend permission.
+
+```solidity
+function getCurrentPeriod(SpendPermission memory spendPermission) public view returns (PeriodSpend memory);
+```
+## Coinbase Smart Wallet clients
+
+### Using Viem
+[Viem](https://viem.sh/) is a TypeScript interface for Ethereum that provides low-level stateless primitives for interacting with Ethereum.
+
+- **[Getting Started with Viem](https://viem.sh/docs/getting-started.html)** 
+
+- **[Getting Started with Account Abstraction](https://viem.sh/account-abstraction#getting-started-with-account-abstraction)** 
+   Viem provides support for account abstraction clients, including specific support for Coinbase Smart Wallet and paymasters.
+
+### Using OnchainKit
+[OnchainKit](https://onchainkit.xyz/) is a collection of React components and TypeScript utilities that help developers quickly build
+onchain applications.
+
+### Using Wagmi
+[Wagmi](https://wagmi.sh/) is a collection of React Hooks that facilitate development of blockchain frontends.
+
+-  **[Getting Started with Wagmi](https://wagmi.sh/react/getting-started)** 
+- **[coinbaseWallet](https://wagmi.sh/react/api/connectors/coinbaseWallet)** 
+
+
+## Coinbase Wallet API
+
+### `wallet_fetchPermissions`
```

Comment:
Need to declare the endpoint this is accessible at and also emphasize that this is JSON-RPC style